src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -663,11 +668,60 @@ public boolean isWithinSolidBlock() {
     // Health
 
     @Override
-    public <T extends Projectile> T launchProjectile(Class<? extends T> projectile,
-        Vector velocity) {
-        T entity = world.spawn(getEyeLocation(), projectile);
-        entity.setVelocity(velocity);
-        return entity;
+    public <T extends Projectile> T launchProjectile(Class<? extends T> clazz, Vector vector) {
+        float offset = 0.0F;
+        float velocity = 1.5F;
+        T projectile = launchProjectile(clazz, vector, offset, velocity);
+        return projectile;
+    }
+
+    /**
+     * Launches a projectile from this entity.
+     *
+     * @param clazz the projectile class
+     * @param vector TODO: document this parameter
+     * @param offset TODO: document this parameter
+     * @param velocity the velocity for the first flight tick
+     * @param <T> the projectile class
+     * @return the launched projectile
+     */
+    public <T extends Projectile> T launchProjectile(Class<? extends T> clazz, Vector vector,
+            float offset, float velocity) {
+        if (vector == null) {
+            vector = getVelocity();
+        }
+
+        T projectile = throwProjectile(clazz, getEyeLocation().clone(), vector, offset, velocity);
+        projectile.setShooter(this);
+        return projectile;
+    }
+
+    protected <T extends Projectile> T throwProjectile(Class<? extends T> type, Location location,
+            Vector originalVector, float offset, float velocity) {
+        double k = Math.toRadians(-1);
```

Comment:
Could you rename these k variables to something more descriptive?

Reply:
I made this variable to avoid using `Math.toRadians` for both pitch and yaw. In retrospect, there's a lot of stuff that can be simplified with this math (considering `cos(-x) = cos(x)`). I would definitely clean this up.

Reply:
Done: Calling this one `negDegreesPerRadian` and the other one `magnitude`.